### ISSUE TITLE
Convert all doc-strings to Google Style

### ISSUE THREAD

**Issue by ns-rse** [claritychallenge/clarity#18]
Docstrings are currently written in Python's [ReSructured Text](https://www.writethedocs.org/guide/writing/reStructuredText/).

An easier to read and more flexible docstring style is the [Google Style docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) style which [Sphinx](https://www.sphinx-doc.org/en/master/) supports additional sections and renders with Sphinx via the [napoleon](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/sphinxcontrib.napoleon.html) Sphinx extension.

In particular including examples is very useful.

**Comment by ns-rse** [comment]
This is the only outstanding issue against CPC2 - pyclarity 0.3.0, whilst not complete I'm wondering if a release is required to align with the challenge?

**Comment by jonbarker68** [comment]
Yes, I think so. I was just waiting for the Cadenza recipes to be final. If it's OK with @groadabike I'd propose that we close any open PRs and make a release 0.3.0 on Friday.
